From 3c23c7a8ae7ecc3c5780fe575c53da098bb9f7e6 Mon Sep 17 00:00:00 2001
From: Tsuchiya Yuto <kitakar@gmail.com>
Date: Mon, 13 Jul 2020 11:02:08 +0900
Subject: [PATCH 2/7] ACPI / PMIC: Add i2c address for intel_pmic_bytcrc

On Surface 3, when executing intel_soc_pmic_exec_mipi_pmic_seq_element(),
I recieve the following error:

    [ 7196.356682] intel_soc_pmic_exec_mipi_pmic_seq_element: Not implemented
    [ 7196.356686] intel_soc_pmic_exec_mipi_pmic_seq_element: i2c-addr: 0x6e reg-addr 0x57 value 0x63 mask 0xff

Surface 3 is an Atom Cherry Trail device and has PMIC device "INT33FD",
which is called "Crystal Cove PMIC". It seems that usually, _HRV of the
PMIC is 0x03 if device is Cherry Trail. However, DSDT of Surface 3 describes
the value is 0x02:

    Name (_HRV, 0x02)  // _HRV: Hardware Revision

due to this _HRV value, intel_pmic_bytcrc will be used as PMIC driver,
not intel_pmic_chtcrc.

It seems that the PMIC device "INT33FD" is located at i2c address 0x6e
on Surface 3. The latter driver defines pmic_i2c_address as 0x6e:

    static struct intel_pmic_opregion_data intel_chtcrc_pmic_opregion_data = {
        .pmic_i2c_address = 0x6e,
    };

but the former driver doesn't define the address, resulting in function
intel_soc_pmic_exec_mipi_pmic_seq_element() not working with errors
mentioned above.

This commit adds the i2c address 0x6e to the intel_pmic_bytcrc driver.

TODO: Not sure if the address is the same value for all devices using
      the driver.

Reference: cc0594c4b0ef ("ACPI / PMIC: Add i2c address for thermal control")
Signed-off-by: Tsuchiya Yuto <kitakar@gmail.com>
---
 drivers/acpi/pmic/intel_pmic_bytcrc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/acpi/pmic/intel_pmic_bytcrc.c b/drivers/acpi/pmic/intel_pmic_bytcrc.c
index f2cdb5e4704cf..61c0778ba6b42 100644
--- a/drivers/acpi/pmic/intel_pmic_bytcrc.c
+++ b/drivers/acpi/pmic/intel_pmic_bytcrc.c
@@ -282,6 +282,7 @@ static struct intel_pmic_opregion_data intel_crc_pmic_opregion_data = {
 	.power_table_count= ARRAY_SIZE(power_table),
 	.thermal_table	= thermal_table,
 	.thermal_table_count = ARRAY_SIZE(thermal_table),
+	.pmic_i2c_address = 0x6e,
 };
 
 static int intel_crc_pmic_opregion_probe(struct platform_device *pdev)
-- 
2.27.0

